pkg/controller/awselb/aws_elb_controller.go

```diff
@@ -265,11 +279,24 @@ func (c *Controller) syncMachine(key string) error {
 
 	region := coMachineSetSpec.ClusterHardware.AWS.Region
 	mLog.Debugf("Obtaining AWS clients for region %q", region)
-	client, err := clustopaws.NewClient(c.kubeClient, coMachineSetSpec, machine.Namespace, region)
+	client, err := c.clientBuilder(c.kubeClient, coMachineSetSpec, machine.Namespace, region)
+	if err != nil {
+		return err
+	}
+
+	status, err := controller.AWSMachineProviderStatusFromClusterAPIMachine(machine)
 	if err != nil {
 		return err
 	}
 
+	if status.LastELBSync != nil {
+		delta := time.Now().Sub(status.LastELBSync.Time)
+		if delta < elbResyncDuration {
+			mLog.WithField("delta", delta).Debugf("no resync needed")
+			return nil
+		}
+	}
+
```

Comment:
We're missing the part where `LastELBSync` is set after successfully syncing the ELB. We should probably add a check for that in the unit tests.

Reply:
Yup, that actually exposes a problem with what we just decided to do in the aws actuator where we just rebuild status from nothing and assume nothing else is writing to it. I will get that fixed as well.